v-next/hardhat-utils/src/request.ts

```diff
@@ -84,7 +91,10 @@ export async function getRequest(
       ...baseRequestOptions,
     });
   } catch (e) {
-    ensureError(e);
+    ensureError<NodeJS.ErrnoException>(e);
```

Comment:
As mentioned somewhere else in the PR, this is actually unsafe.

Reply:
Done [here](https://github.com/NomicFoundation/hardhat/pull/5580/commits/7229bd0a281274f58361b6468a5957a5d29ca124)